### PR TITLE
*: fix data race when defer cancel() is called

### PR DESCRIPTION
**Pull request by tiancaiamao**
defer cancel() was introduced in this PR
https://github.com/pingcap/tidb/pull/3010

### CODE REVIEW

**Comment by tiancaiamao** [comment]
Try to fix CI
PTAL @coocood

**Comment by tiancaiamao** [comment]
PTAL @hhkbp2 @hanfei1991

File: context/context.go [hunk 1]
```diff
@@ -32,6 +32,9 @@ type Context interface {
 	// Txn returns the current transaction which is created before executing a statement.
 	Txn() kv.Transaction
 
+	// GoCtx returns the standard context.Context that bind with current transaction.
```
Comment (hhkbp2) [inline context/context.go]: s/that bind/which is bound/

**Comment by hhkbp2** [comment]
Rest LGTM

**Comment by zimulala** [comment]
LGTM

**Pull request merged**